modules/vaos/app/services/vaos/appointment_requests_service.rb

```diff
@@ -74,11 +82,11 @@ def filter_cc_appointment_data(request)
       request
     end
 
-    def url(id = nil)
+    def url(id = nil, type = 'appointments')
       if id
         url + "/system/var/id/#{id}"
       else
-        "/var/VeteranAppointmentRequestService/v4/rest/appointment-service/patient/ICN/#{user.icn}/appointments"
+        "/var/VeteranAppointmentRequestService/v4/rest/appointment-service/patient/ICN/#{user.icn}/#{type}"
```

Comment:
The if/else here seems to do two pretty different things on each logic path. Its clear one path generates an appointment endpoint URL -- the other not so clear. Maybe `system/var/id/:id` is another URL for an appointment but its pretty obtuse without specific knowledge.

This code plus the `if` in the form object makes it seem like there are really two different flows that are pushed together into this one process. Wondering if there's a higher level refactor that makes everything simpler.

Reply:
it's kind of a recursive if statement. I think I got a little code golf crazy here. I thought it was elegant, but not necessarily readable. I'll certainly be refactoring this. I'm actually looking to move most of the services logic into a base services class with url helper methods, etc.